frontend/src/scenes/plugins/Plugins.tsx

```diff
@@ -6,72 +6,80 @@ import { InstalledTab } from 'scenes/plugins/tabs/installed/InstalledTab'
 import { useActions, useValues } from 'kea'
 import { userLogic } from 'scenes/userLogic'
 import { pluginsLogic } from './pluginsLogic'
-import { Spin, Tabs, Tag } from 'antd'
-import { OptInPlugins } from 'scenes/plugins/optin/OptInPlugins'
+import { Alert, Tabs } from 'antd'
 import { PageHeader } from 'lib/components/PageHeader'
 import { PluginTab } from 'scenes/plugins/types'
 import { AdvancedTab } from 'scenes/plugins/tabs/advanced/AdvancedTab'
 import { canGloballyManagePlugins, canInstallPlugins, canViewPlugins } from './access'
 
 export function Plugins(): JSX.Element | null {
     const { user } = useValues(userLogic)
+    const { userUpdateRequest } = useActions(userLogic)
     const { pluginTab } = useValues(pluginsLogic)
     const { setPluginTab } = useActions(pluginsLogic)
     const { TabPane } = Tabs
 
-    if (!user) {
-        return <Spin />
-    }
+    useEffect(() => {
+        if (user) {
+            if (!canViewPlugins(user.organization)) {
+                window.location.href = '/'
+                return
+            }
+            if (!user.flags.has_checked_out_plugins) {
+                userUpdateRequest({ user: { flags: { has_checked_out_plugins: true } } })
+                return
+            }
+        }
+    }, [user])
 
-    if (!canViewPlugins(user.organization)) {
-        useEffect(() => {
-            window.location.href = '/'
-        }, [])
+    if (!user || !canViewPlugins(user.organization)) {
```

Comment:
Is `!user` possible here? AFAIK scene logic does verify user exists at this point. If not, we should avoid adding code to cover impossible scenarios and use `user!`

Reply:
Well, it's not a pure component, uses Kea, so IMO better to have a practically free non-null check.

Reply:
Actually, made it not use Kea for this, so addressed.